dvc/stage/__init__.py

```diff
@@ -306,7 +306,12 @@ def remove(self, force=False, remove_outs=True, purge=True):
     @rwlocked(read=["deps"], write=["outs"])
     def reproduce(self, interactive=False, **kwargs):
         if not (kwargs.get("force", False) or self.changed()):
-            logger.info("Stage '%s' didn't change, skipping", self.addressing)
+            if not isinstance(self, PipelineStage) and self.is_data_source:
+                logger.info("'%s' didn't change, skipping", self.addressing)
```

Comment:
Should not this be `foo didn't change` rather `foo.dvc`?

Reply:
I thought the same @skshetry but changed my mind on the argument that as `.dvc files` maintain the updates in the original files hence mentioning `foo.dvc` rather than `foo` sounded logical to me.

Also `.dvc files` are kind of representatives of the original files so using them in the place of original ones should be good IMHO.